session.go

```diff
@@ -78,15 +78,21 @@ type ServerCommandHandler func(ctx context.Context, sess Session) error
 type ConnectOption func(*connectConfig)
 
 type clientInfo struct {
-	Type    string
-	Version string
+	Type     string
+	Version  string
+	Comments []string
 }
 
 var bannedUAchar = regexp.MustCompile("[^!#$%&'*+-.^_`|~0-9a-zA-Z]")
 
 // Formats client info as a well-formed user agent string
 func (c *clientInfo) ToUserAgent() string {
-	return sanitizeUserAgentString(c.Type) + "/" + sanitizeUserAgentString(c.Version)
+	comments := make([]string, 0, len(c.Comments))
+	for _, comment := range c.Comments {
+		comments = append(comments, fmt.Sprintf(" (%s)", comment))
+	}
+
+	return sanitizeUserAgentString(c.Type) + "/" + sanitizeUserAgentString(c.Version) + strings.Join(comments, " ")
```

Comment:
You've got an extra space in either your Join or your Sprintf. Probably be easiest to just change your Join to use `""` since you'll want that leading space from the `Sprintf` anyway.